src/ts/waterfall/details-overlay/svg-details-overlay.ts

```diff
@@ -51,10 +65,14 @@ export function createRowInfoOverlay(overlay: OpenOverlay, y: number, detailsHei
     y,
   });
 
+  const body = createDetailsBody(requestID, detailsHeight, overlay.entry);
   const closeBtn = createCloseButtonSvg(y);
-  closeBtn.addEventListener("click", () => overlay.onClose(overlay.index));
+  closeBtn.addEventListener("click", () => {
+    overlay.onClose(overlay.index);
+    body.removeEventListener("click", onTabDataCopyClick);
+  });
+  body.addEventListener("click", onTabDataCopyClick);
```

Comment:
use event delegation as we can't bind callbacks before the tab has been added to the DOM, this also allows us to clean up the event listeners when it gets removed.